src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1025,11 +1025,21 @@ private HttpRequestMessage CreateRequest<TResult>(JsonOperationContext ctx, Serv
         {
             var request = command.CreateRequest(ctx, node, out url);
 
-            request.RequestUri = new Uri(url);
+            var builder = new UriBuilder(url);
 
-            if (!request.Headers.Contains(Constants.Headers.ClientVersion))
-                request.Headers.Add(Constants.Headers.ClientVersion, ClientVersion);
+            if (command is IRaftCommand raftCommand)
+            {
+                Debug.Assert(raftCommand.RaftUniqueRequestId != null, $"Forget to create an id for {command.GetType()}?");
+
+                var raftRequestString = "raft-request-id=" + raftCommand.RaftUniqueRequestId;
+                builder.Query = builder.Query?.Length > 1 ? $"{builder.Query.Substring(1)}&{raftRequestString}" : raftRequestString;
```

Comment:
Why are you skipping the first char in the existing `Query` ?

Reply:
followed this : https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder.query?view=netframework-4.8